File: src/api/schemas/backup.py
"""运行时数据备份端点的请求 / 响应 schema。"""

from pydantic import BaseModel, Field


class CreateBackupRequest(BaseModel):
    # 要备份的类别（{A,B,C,E,F,K} 子集）；默认全选。空列表非法（路由校验）
    categories: list[str] = Field(default_factory=lambda: ["A", "B", "C", "E", "F", "K"])


class BackupSnapshot(BaseModel):
    name: str
    timestamp: str
    created_at: str = ""
    include_vectors: bool | None = None
    file_count: int
    zip_bytes: int
    category_stats: dict = {}


class BackupListResponse(BaseModel):
    items: list[BackupSnapshot]


class RestoreResponse(BaseModel):
    restored: int        # 还原的文件数
    message: str         # 给前端展示的提示（含重启建议）

File: frontend/src/types/backup.ts
// 运行时数据备份（/admin/backup/*）类型，与后端 src/api/schemas/backup.py 对齐。

export type BackupSnapshot = {
  name: string
  timestamp: string
  created_at: string
  include_vectors: boolean | null
  file_count: number
  zip_bytes: number
  category_stats: Record<string, { files: number; bytes: number }>
}

export type BackupListResponse = {
  items: BackupSnapshot[]
}

export type RestoreResponse = {
  restored: number
  message: string
}
